lib/test_formulae.rb

```diff
@@ -97,33 +97,48 @@ def download_artifact_from_previous_run!(name)
         response = GitHub::API.open_rest(url)
         return if response.fetch("total_count").zero?
 
-        wanted_artifact = response.fetch("artifacts").find { |artifact| artifact.fetch("name") == name }
+        artifacts = response.fetch("artifacts")
+        wanted_artifact = artifacts.find { |artifact| artifact.fetch("name") == name }
+        wanted_artifact ||= artifacts.find { |artifact| artifact.fetch("name") == "event_payload" }
         return if wanted_artifact.blank?
 
-        ohai "Downloading #{name} from #{previous_github_sha}"
+        wanted_artifact_name = wanted_artifact.fetch("name")
+        cached_event_json&.unlink if wanted_artifact_name == "event_payload"
+
+        ohai "Downloading #{wanted_artifact_name} from #{sha}"
         download_url = wanted_artifact.fetch("archive_download_url")
 
         artifact_cache.mkpath
         artifact_cache.cd do
           GitHub.download_artifact(download_url, run_id)
         end
+
+        return if wanted_artifact_name == name
+
+        download_artifact_from_previous_run!(name)
```

Comment:
Honestly some comments here with the intended flow could be useful. I was confused at first as I thought this would have downloaded a bottle from a different run from the event payload.

Reply:
Yup, that's exactly how this works. Will add some comments.

Reply:
Added in 8b1655066fdc15b91dd2a203a87bea3b5944c1a9.